src/cljs/org/broadinstitute/firecloud_ui/main.cljs

```diff
@@ -327,66 +336,130 @@
                  " for more information."])}))
 
 
+(react/defc ConfigLoader
+  {:render
+   (fn [{:keys [state]}]
+     [:div {:style {:padding "40px 0"}}
+      (if (:error? @state)
+        [:div {:style {:color (:exception-state style/colors)}}
+         "Error loading configuration. Please try again later."]
+        [comps/Spinner {:text "Loading configuration..."}])])
+   :component-did-mount
+   (fn [{:keys [props state]}]
+     ;; Use basic ajax call here to bypass authentication.
+     (utils/ajax {:url "/config.json"
+                  :on-done (fn [{:keys [success? get-parsed-response]}]
+                             (if success?
+                               (do
+                                 (reset! config/config (get-parsed-response))
+                                 ((:on-success props)))
+                               (swap! state assoc :error true)))}))})
+
+
+(react/defc UserStatus
+  {:render
+   (fn [{:keys [state]}]
+     [:div {:style {:padding "40px 0"}}
+      (case (:error @state)
+        nil [comps/Spinner {:text "Loading user information..."}]
+        :not-active [:div {:style {:color (:exception-reds style/colors)}}
+                     "Thank you for registering. Your account is currently inactive."
+                     " You will be contacted via email when your account is activated."]
+        [:div {:style {:color (:exception-state style/colors)}}
+         "Error loading user information. Please try again later."])])
+   :component-did-mount
+   (fn [{:keys [props state]}]
+     (utils/ajax-orch "/me"
+                      {:on-done (fn [{:keys [success? status-code]}]
+                                  (if success?
+                                    ((:on-success props))
+                                    (case status-code
+                                      403 (swap! state assoc :error :not-active)
+                                      ;; 404 means "not yet registered"
+                                      404 ((:on-success props))
+                                      (swap! state assoc :error true))))}
```

Comment:
Nope this is right. Basically, if successful or 404, proceed, since that either means the user is good to go or they are not yet registered. Suggestions for improving clarity welcome.